### ISSUE TITLE
False positive E122

### ISSUE THREAD

**Issue by thisfred** [PyCQA/pycodestyle#678]
This:

```python
dictionary = {
    'key1':
        len('bar'),
    'key2': [
        'value',
    ],
}
```

results in:

```
test.py:6:5: E122 continuation line missing indentation or outdented
```

so the closing square bracket is misidentified as being incorrectly indented.

Which is weird. Weirder still: the warning disappears if the value of `key1` is changed to just `'bar'`, so I think it has something to do with the parentheses. The warning *also* disappears if the first key and value are put on the same line.

**Comment by zeusttu** [comment]
This error also occurs if `key2` contains a dict insterad of a list. Furthermore, I find that the error goes away if the closing bracket of `key2` is indented by an extra step, so the following is considered ok:

    {
        "key1":
            len("bar"),
        "key2": [
            "value",
            ],
    },

I hope this information helps in debugging the problem.

**Comment by zeusttu** [comment]
I also discovered that it only occurs if `key1` in the above examples is specified first. The following does produce the error:

    {
        "key0": [
            "value",
        ],
        "key1":
            len("bar"),
        "key2": [
            "value",
        ],
    }
    }

**Comment by zeusttu** [comment]
Looking at the code, I find that the line number of `"key1":` gets appended to the `open_rows[depth]` variable, and is not removed from it again when the value belonging to `key1` ends. I think it should be removed from that list in the case outlined in this issue. I am however not enough "at home" in Pycodestyle's codebase to figure out exactly under what conditions it should or should not be removed from that list, so I will leave the implementation to people more involved with the project.

**Comment by asottile** [comment]
Duplicate of #408
